systemservices/marketplace/src/types/schema/definition.json

```diff
@@ -0,0 +1,246 @@
+{
+  "$schema": "http://json-schema.org/draft-06/schema#",
```

Comment:
are we sure that this definition is still valid with the one we have in the core ?
I see sid for example:
Core:
```
Sid string `yaml:"sid" json:"sid,omitempty" validate:"omitempty,printascii,max=63,domain"`
```
Here:
```
"sid": {
      "type": "string",
      "maxLength": 39,
      "pattern": "^[0-9a-zA-Z\\._-]+$"
 },
```

This is just one example I looked at but we might have more and I think now the core only should be responsible for this validation. Not this service (or this service should have a way to delegate to the core with an api or another service that the core uses with service composition). I really don't think this is a good idea, I would prefer to return invalid data then have a good error when we deploy than having invalid errors and miss some versions.

Reply:
I update the validation of the service definition.
I think this validation is not perfect but already cover 90% of the definition and 100% of the structure. I think it's anyway a good thing to make sure the structure is correct.
Tell me if your ok to keep it or completely discard it.

Reply:
I just feel this is really risky because we will probably update the structure in the core multiple time and this validation will not possibly keep up with that.
I think we should consider this definition as valid by default and let the core handle this.
One solution could be to provide an api for validation either with service composition or a core api.

Reply:
I'm not 100% against it, I'm ok to keep it but with a big warning and I think better to run with no validation for now and do something nice than run on a partial validation that will be really hard to maintain

Reply:
> I just feel this is really risky because we will probably update the structure in the core multiple time and this validation will not possibly keep up with that.
I think we should consider this definition as valid by default and let the core handle this.
One solution could be to provide an api for validation either with service composition or a core api.

i totally agree. But what to do until there is a validation api? And even, the validation api will be available on the core api but this is a service...

> I'm not 100% against it, I'm ok to keep it but with a big warning and I think better to run with no validation for now and do something nice than run on a partial validation that will be really hard to maintain

i'm very afraid that the website can be broken because of wrong service structure...